JUnit failure trace view: Stracktrace garbled (org.junit.jupiter truncated from message)

I have to create my own assertion error message in an advanced JUnit.
When I run the JUnit with Maven, I get the proper output:
```
[ERROR] testReadProject  Time elapsed: 0.407 s  <<< FAILURE!
java.lang.AssertionError:
Expected
<[Dependency {groupId=org.apache.maven, artifactId=maven-core, version=3.6.1, type=jar}, Dependency {groupId=io.github.m-m-m, artifactId=mmm-code-base, version=0-SNAPSHOT, type=jar}, Dependency {groupId=org.mockito, artifactId=mockito-junit-jupiter, version=4.9.0, type=jar}, Dependency {groupId=org.junit.jupiter, artifactId=junit-jupiter, version=5.9.1, type=jar}, Dependency {groupId=org.assertj, artifactId=assertj-core, version=3.23.1, type=jar}, Dependency {groupId=ch.qos.logback, artifactId=logback-classic, version=1.4.5, type=jar}]>
to contain
<[Dependency {groupId=io.github.m-m-m, artifactId=mmm-code-base, version=0-SNAPSHOT, type=jar}, Dependency {groupId=org.apache.maven, artifactId=maven-core, version=3.6.1, type=jar}, Dependency {groupId=org.assertj, artifactId=assertj-core, version=3.23.1, type=jar}, Dependency {groupId=org.junit.jupiter, artifactId=junit-jupiter, version=5.9.0, type=jar}, Dependency {groupId=org.mockito, artifactId=mockito-junit-jupiter, version=4.9.0, type=jar}, Dependency {groupId=ch.qos.logback, artifactId=logback-classic, version=1.4.5, type=jar}]>
but could not find
<[Dependency {groupId=org.junit.jupiter, artifactId=junit-jupiter, version=5.9.0, type=jar}]>
 and did not expect
<[Dependency {groupId=org.junit.jupiter, artifactId=junit-jupiter, version=5.9.1, type=jar}]>
        at io.github.mmm.code.java.maven.impl.MavenBridgeImplTest.verifyDependencies(MavenBridgeImplTest.java:133)
        at io.github.mmm.code.java.maven.impl.MavenBridgeImplTest.testReadProject(MavenBridgeImplTest.java:91)
```

However, Eclipse does make evil black magic here and simply removes lines from the exception message that contain brackets:
<img width="1271" alt="Eclipse-JUnit-stracktrace-garbled" src="https://user-images.githubusercontent.com/1923119/211167021-422f9ed9-4a86-466c-b01a-f4fa52565698.png">

As a result the important information is missing and the result is rendered useless.
It would be awesome if you could consider adapting this magic that truncates stacktraces.
I also created a minimal test-case and found out that the keyword `org.junit.jupiter` is triggering this effect.
While it makes fully sense to remove such lines from the actual stacktrace call hierarchy, it IMHO does not make sense to remove it from the actual exception message.
Here is my test-case:
```
import org.junit.jupiter.api.Test;

public class MyTest {
  @Test
  public void test() {
    throw new AssertionError("Eclipse truncates lines with\n[org.junit.jupiter]\nfrom stracktraces.");
  }
}
```

--- a/org.eclipse.jdt.junit/src/org/eclipse/jdt/internal/junit/ui/TextualTrace.java
+++ b/org.eclipse.jdt.junit/src/org/eclipse/jdt/internal/junit/ui/TextualTrace.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2005, 2017 IBM Corporation and others.
+ * Copyright (c) 2005, 2023 IBM Corporation and others.
  *
  * This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License 2.0
@@ -119,11 +119,15 @@ public class TextualTrace {
 		String line;
 		String[] patterns = filterPatterns;
 		boolean firstLine= true;
+		boolean stackFrameSeen= false;
 		try {
 			while ((line= bufferedReader.readLine()) != null) {
-				if (firstLine || !filterLine(patterns, line))
+				if (firstLine || !stackFrameSeen || !filterLine(patterns, line))
 					printWriter.println(line);
 				firstLine= false;
+				if (!(stackFrameSeen)) {
+					stackFrameSeen= isAStackFrame(line.replace('\t', ' '));
+				}
 			}
 		} catch (IOException e) {
 			return stackTrace; // return the stack unfiltered

